Assets/Scripts/Edges/EdgeGameObjectFactory.cs

```diff
@@ -1,76 +1,66 @@
 ﻿using System.Collections.Generic;
 using System.Linq;
-using Assets.Scripts.Common;
 using Assets.Scripts.Model;
 using UnityEngine;
+using UnityEngine.UI;
 
 namespace Assets.Scripts.Edges
 {
   public class EdgeGameObjectFactory : MonoBehaviour
   {
-    private const float _additionalColliderLength = 0.1f;
-    private const float _edgeColliderRadius = 0.09f;
-    public GameObject edgePrefab;
+    const float c_edgeThicknessMultiplier = 0.1f;
+    public GameObject labelPrefab;
     public LineFactory lineFactory;
+    public MaterialCache _materialCache;
+    public void Start()
+    {
+      _materialCache = FindObjectOfType<MaterialCache>();
+    }
 
-    public void CreateGameObjectEdgesFor(IList<Edge> edges, bool labelVisibility)
+    public List<(SegmentGroup, GameObject)> CreateGameObjectEdgesFor(IList<Edge> edges, bool labelVisibility)
```

Comment:
This method is quite long and it does many things. I would like it to be split up into private methods in the future to help clarity

Reply:
Yup, I didn't bother as I expect to make changes to it. The whole game object creation is getting complicated and I am considering builders.